src/Tracy/Dumper.php

```diff
@@ -415,26 +415,31 @@ public static function encodeString($s, $maxLength = NULL)
 			$table["\t"] = '\t';
 		}
 
-		if (preg_match('#[^\x09\x0A\x0D\x20-\x7E\xA0-\x{10FFFF}]#u', $s) || preg_last_error()) {
+		if (function_exists('mb_substr')) {
+			if ($maxLength && strlen($s) > $maxLength) {
+				$s = mb_substr($tmp = $s, 0, $maxLength, 'UTF-8');
+				$shortened = $s !== $tmp;
+			} else {
+				$shortened = false;
+			}
+			if (preg_match('#[^\x09\x0A\x0D\x20-\x7E\xA0-\x{10FFFF}]#u', $s) || preg_last_error()) {
+				$s = strtr($s, $table);
+			}
+		} elseif (preg_match('#[^\x09\x0A\x0D\x20-\x7E\xA0-\x{10FFFF}]#u', $s) || preg_last_error()) {
 			if ($shortened = ($maxLength && strlen($s) > $maxLength)) {
 				$s = substr($s, 0, $maxLength);
 			}
 			$s = strtr($s, $table);
 
 		} elseif ($maxLength && $s !== '') {
-			if (function_exists('mb_substr')) {
-				$s = mb_substr($tmp = $s, 0, $maxLength, 'UTF-8');
-				$shortened = $s !== $tmp;
-			} else {
-				$i = $len = 0;
-				do {
-					if (($s[$i] < "\x80" || $s[$i] >= "\xC0") && (++$len > $maxLength)) {
-						$s = substr($s, 0, $i);
-						$shortened = TRUE;
-						break;
-					}
-				} while (isset($s[++$i]));
-			}
+			$i = $len = 0;
+			do {
+				if (($s[$i] < "\x80" || $s[$i] >= "\xC0") && (++$len > $maxLength)) {
+					$s = substr($s, 0, $i);
+					$shortened = TRUE;
+					break;
+				}
+			} while (isset($s[++$i]));
```

Comment:
This code block should be only done when the string was not already shortened by mb_substr, i.e. 

~~~php
} elseif ($maxLength && $s !== '') {
~~~

should be changed to 

~~~php
} elseif ($maxLength && $s !== '' && !function_exists('mb_substr')) {
~~~